src/script.c

```diff
@@ -501,3 +504,36 @@ void InitRamScript_NoObjectEvent(u8 *script, u16 scriptSize)
     InitRamScript(script, scriptSize, MAP_GROUP(UNDEFINED), MAP_NUM(UNDEFINED), NO_OBJECT);
 #endif //FREE_MYSTERY_EVENT_BUFFERS
 }
+
+#define WALK_AWAY_SIGNPOST_FRAMES 6
+
+void SetWalkingIntoSignVars(void)
+{
+    gWalkAwayFromSignpostTimer = WALK_AWAY_SIGNPOST_FRAMES;
+    sMsgBoxIsCancelable = TRUE;
+}
+
+bool32 IsMsgSignPost(void)
+{
+    return sMsgIsSignPost;
+}
+
+void ResetFacingNpcOrSignPostVars(void)
+{
+    sMsgIsSignPost = FALSE;
+}
+
+void MsgSetSignPost(void)
+{
+    sMsgIsSignPost = TRUE;
+}
+
+void ClearMsgBoxCancelableState(void)
+{
+    sMsgBoxIsCancelable = FALSE;
+}
+
+bool32 CanWalkAwayToCancelMsgBox(void)
+{
+    return sMsgBoxIsCancelable;
+}
```

Comment:
I think this set of functions is a bit redundant. It should be self explanatory what it means when a specific value is set/cleared without a function that tells you what is done.